### DISCUSSION TITLE
sql exception

### DISCUSSION THREAD

**Discussion by JohnRDOrazio** [llaville/php-compat-info#377] [Q&A] [answered]
I have successfully used the `phpcompatinfo` script installed globally with composer in the past couple years. However when running it today on a project folder with `phpcompatinfo analyser:run ./src`, I got the following exception:

> * An exception occurred while executing a query: SQLSTATE[HY000]: General error: 1 no such column: t0.deprecated_since from file "~/.config/composer/vendor/doctrine/dbal/src/Driver/API/SQLite/ExceptionConverter.php" at line 83
>
> ! [NOTE] Issue found by phpCompatInfo version 7.1.4@9875282 with DB version 6.12.0@23c21f6

I tried `composer global update` but still no dice, same exception occurs when running `phpcompatinfo analyser:run`.

`phpcompatinfo diagnose` shows everything as green:

Requirements Checker
====================

 > Running PHP CompatInfo with PHP 8.3.10 on Linux 5.15.153.1-microsoft-standard-WSL2
 > PHP is using the following php.ini file:
 /etc/php/8.3/cli/php.ini

 > Checking PHP CompatInfo requirements:
   [x] PHP version must be at least 8.1.0
   [x] pdo_sqlite extension must be available
   [x] Check if source "/home/johnrdorazio/.cache/bartlett/prod/compatinfo-db.sqlite" can be reached
   [x] Check if tables exists in database
   [x] Check if database contains at least one distribution platform

 ? Environment

 ? Parameters
    >  [kernel.environment] prod
    >  [kernel.debug]
    >  [kernel.home_dir] /home/johnrdorazio
    >  [kernel.project_dir] /home/johnrdorazio/.config/composer/vendor/bartlett/php-compatinfo
    >  [kernel.cache_dir] /home/johnrdorazio/.cache/bartlett/prod
    >  [kernel.build_dir] /home/johnrdorazio/.cache/bartlett/prod
    >  [kernel.logs_dir] /home/johnrdorazio/.cache/bartlett/prod
    >  [kernel.vendor_dir] /home/johnrdorazio/.config/composer/vendor/bartlett/php-compatinfo/vendor
    >  [kernel.container_class] Bartlett_CompatInfo_Application_Kernel_ConsoleKernelProdContainer
    >  [compat_info_db.version] 6.6.0@1f2c77e
    >  [compat_info_db.proxy_dir] /home/johnrdorazio/.cache/bartlett/prod/php-compatinfo-db/6.6.0@1f2c77e/proxies

 [OK] Your system is ready to run the application.

 ! [NOTE] phpCompatInfo version 7.1.4@9875282 DB version 6.12.0@23c21f6

------------------------

My default PHP version is 8.3. What steps should I take to fix this?

**Comment by llaville** [comment]
Hello @JohnRDOrazio 

The answer is pretty simple.  Thanks to sharing `diagnose` results. I need to improve it, in this context to display what to do !

So, with your versions 

> ! [NOTE] phpCompatInfo version 7.1.4@9875282 DB version 6.12.0@23c21f6

DB version is 6.12.0 (the latest version available on today date). 
Cool but this [release](https://github.com/llaville/php-compatinfo-db/releases/tag/6.12.0) has introduced a change in database structure to be able to display all deprecations on each extension elements.

As you've installed and currently use another version (6.6.0) identified by these lines

> [compat_info_db.version] 6.6.0@1f2c77e
> [compat_info_db.proxy_dir] /home/johnrdorazio/.cache/bartlett/prod/php-compatinfo-db/6.6.0@1f2c77e/proxies 

I suggest to rebuild the database contents, with following command : 

```shell
phpcompatinfo db:new --force
```

And checks again with `diagnose` command that all DB references (6.12.0@23c21f6) are correct. Something like 

```
    >  [compat_info_db.version] 6.12.0@23c21f6
    >  [compat_info_db.proxy_dir] /home/devilbox/.cache/bartlett/prod/php-compatinfo-db/6.12.0@23c21f6/proxies


 [OK] Your system is ready to run the application.


 ! [NOTE] phpCompatInfo version 7.2.x-dev@3ad4d00 DB version 6.12.0@23c21f6
```

I'm currently showing results with de 7.2.0 version still under development, but it should be same with 7.1.4

Hope I've answered your question and expectations ?

**Reply by llaville** [reply]
On my TODO list with : https://github.com/llaville/php-compatinfo-db/issues/140

**Comment by JohnRDOrazio** [comment]
I tried to run `phpcompatinfo db:new --force`, however even this command gives the same error:
```
 [ERROR] An exception occurred while executing a query: SQLSTATE[HY000]: General error: 1 table ini_entries has no
         column named deprecated_since
```

Running `phpcompatinfo diagnose` after this gives me the following:

```
Requirements Checker
====================

 > Running PHP CompatInfo with PHP 8.3.10 on Linux 5.15.153.1-microsoft-standard-WSL2
 > PHP is using the following php.ini file:
 /etc/php/8.3/cli/php.ini

 > Checking PHP CompatInfo requirements:
   [x] PHP version must be at least 8.1.0
   [x] pdo_sqlite extension must be available
   [x] Check if source "/home/johnrdorazio/.cache/bartlett/prod/compatinfo-db.sqlite" can be reached
   [x] Check if tables exists in database

   [ ] Check if database contains at least one distribution platform

 ? Environment

 ? Parameters
    >  [kernel.environment] prod
    >  [kernel.debug]
    >  [kernel.home_dir] /home/johnrdorazio
    >  [kernel.project_dir] /home/johnrdorazio/.config/composer/vendor/bartlett/php-compatinfo
    >  [kernel.cache_dir] /home/johnrdorazio/.cache/bartlett/prod
    >  [kernel.build_dir] /home/johnrdorazio/.cache/bartlett/prod
    >  [kernel.logs_dir] /home/johnrdorazio/.cache/bartlett/prod
    >  [kernel.vendor_dir] /home/johnrdorazio/.config/composer/vendor/bartlett/php-compatinfo/vendor
    >  [kernel.container_class] Bartlett_CompatInfo_Application_Kernel_ConsoleKernelProdContainer
    >  [compat_info_db.version] 6.6.0@1f2c77e
    >  [compat_info_db.proxy_dir] /home/johnrdorazio/.cache/bartlett/prod/php-compatinfo-db/6.6.0@1f2c77e/proxies


 [ERROR] Your system is not ready to run the application.


Fix the following mandatory requirements (in sequential order):
---------------------------------------------------------------

   * At least one distribution platform should exist. None available. Run "db:init" command to build one.

 ! [NOTE] phpCompatInfo version 7.1.4@9875282 DB version 6.12.0@23c21f6
```

So I tried running `phpcompatinfo db:init`, but this still gives me:

```
[ERROR] An exception occurred while executing a query: SQLSTATE[HY000]: General error: 1 table ini_entries has no
         column named deprecated_since
```

Not sure where to go from here?

**Comment by llaville** [comment, marked as answer]
Remove the cache directory dentified by 'kernel.cache_dir', and retry

**Comment by JohnRDOrazio** [comment]
Ok after removing the cache folder, I was able to `compatinfo-db db:create` and then `compatinfo-db db:init`.

I'm guessing `phpcompatinfo db:<command>` and `compatinfo-db db:<command>` are synonymous?

In any case `phpcompatinfo diagnose` now gives me `>  [compat_info_db.version] 6.12.0@23c21f6`.

And `phpcompatinfo analyse:run ./src` is now working again.

**Reply by llaville** [reply]
> 
> I'm guessing `phpcompatinfo db:<command>` and `compatinfo-db db:<command>` are synonymous?

Yes, these commands act as a bridge 

> 
> In any case `phpcompatinfo diagnose` now gives me `> [compat_info_db.version] 6.12.0@23c21f6`.
> 
> And `phpcompatinfo analyse:run ./src` is now working again.

Glad to help you !
